src/Microsoft.AspNetCore.Mvc.Razor.Host/Internal/ViewComponentTagHelperChunkVisitor.cs

```diff
@@ -131,9 +131,23 @@ private void BuildAttributeDeclarations(TagHelperDescriptor descriptor)
                 $"global::{_context.ViewContextTypeName}",
                 ViewContextVariable);
 
+            var indexerAttributes = descriptor.Attributes.Where(a => a.IsIndexer);
+
             foreach (var attribute in descriptor.Attributes)
             {
+                if (attribute.IsIndexer)
+                {
+                    continue;
+                }
+
                 Writer.WriteAutoPropertyDeclaration("public", attribute.TypeName, attribute.PropertyName);
```

Comment:
Should totally add an overload to this that lets you specify end line or not. Another day though.